aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -250,12 +252,33 @@ def _route(self, event: BaseProxyEvent, cors: Optional[CORSConfig]):
             self._add_cors(event, cors or CORSConfig())
         if self.route.cache_control:
             self._add_cache_control(self.route.cache_control)
-        if self.route.compress and "gzip" in (event.get_header_value("accept-encoding", "") or ""):
+        if (
+            self.route.compress
+            and "gzip" in (event.get_header_value("accept-encoding", "") or "")
+            and self.response.compress is not False
+        ):
+            self._compress()
```

Comment:
could you break this down to reduce cognitive overload?

A private method to control whether to enable compression will be easier to reason. 

e.g., 

`if self._has_compression_enabled(...): self._compress()`

Reply:
I completely agree!! The code is much cleaner now :)